Libplanet.Net/Swarm.BlockSync.cs

```diff
@@ -476,26 +477,47 @@ in completedBlocks.WithCancellation(cancellationToken))
                         $"is unusable for the the existing chain #0 {g}.");
                 }
 
+                if (!(tipCandidate is { }))
+                {
+                    throw new SwarmException(
+                        $"Tip candidate is null.");
+                }
+
                 // NOTE: All blocks in deltaBlocks will have non-null previousHash.
-                var deltaBlocks = new LinkedList<Block<T>>(new[] { tipCandidate });
+                var blockChainSlice = new BlockChainSlice(new[] { tipCandidate.Hash });
                 while (true)
                 {
-                    Block<T> b = deltaBlocks.First();
+                    BlockHash bh = blockChainSlice.Last();
+                    if (!workspace.Store.ContainsBlock(bh))
+                    {
+                        break;
+                    }
+
+                    Block<T> b = workspace.Store.GetBlock<T>(bh);
                     if (b.PreviousHash is { } p && !workspace.ContainsBlock(p))
                     {
-                        deltaBlocks.AddFirst(workspace.Store.GetBlock<T>(p));
+                        blockChainSlice.Add(p);
                     }
                     else
                     {
                         break;
                     }
+
+                    cancellationToken.ThrowIfCancellationRequested();
                 }
 
-                cancellationToken.ThrowIfCancellationRequested();
+                blockChainSlice.Reverse();
```

Comment:
If we construct `blockChainSlice` from `.First()` (as previously), this `.Reverse()` doesn't seem to need. is it right what I think?

Reply:
Previously, the data structure for memorizing target block is `LinkedList<T>` but in this PR, changed to `List<T>`. So I am worried about the performance issue when appending each item to `List<T>`'s index 0.

May we use `LinkedList<T>` again?

Reply:
I've been thinking about it, and changing it to `LinkedList<T>` looks good. I'll roll back to `LinkedList<T>`.

Reply:
ec496acdd1366ea75a3fed132e352d5f47354809 amended